src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/BlobStorageRead.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services;
+
+import com.google.common.io.ByteStreams;
+
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.BlobInputStream;
+import com.microsoft.azure.storage.blob.CloudBlob;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.stream.StreamSupport;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+@Component
+public class BlobStorageRead {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BlobStorageRead.class);
+
+    private final CloudBlobClient cloudBlobClient;
+    private final Consumer<List<PDF>> pdfsConsumer;
+
+    @Autowired
+    public BlobStorageRead(CloudBlobClient cloudBlobClient, Consumer<List<PDF>> pdfsConsumer) {
+        this.cloudBlobClient = cloudBlobClient;
+        this.pdfsConsumer = pdfsConsumer;
+    }
+
+    @Scheduled(fixedDelayString = "${scan.interval}")
+    public void readBlobs() {
+        cloudBlobClient.listContainers().forEach(cloudBlobContainer -> {
+            try {
+                String containerName = cloudBlobContainer.getName();
+                CloudBlobContainer container = cloudBlobClient.getContainerReference(containerName);
```

Comment:
confusing names imho. what's the type of `cloudBlobContainer` variable passed in this lambda?